src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -867,6 +867,61 @@ LogicalResult ONNXSeluOp::inferShapes(
   return success();
 }
 
+LogicalResult ONNXSequenceInsertOp::inferShapes(
+    std::function<void(mlir::Region &)> doShapeInference) {
+  return success();
+}
+
+static LogicalResult verify(ONNXSequenceInsertOp op) {
+  ONNXSequenceInsertOpAdaptor operandAdaptor = ONNXSequenceInsertOpAdaptor(op);
+
+  // These cast should be guaranteed by default verifier
+  auto seqElementType = operandAdaptor.input_sequence()
+                            .getType()
+                            .dyn_cast<mlir::onnxmlir::SeqType>()
```

Comment:
I recommend that we do a "dyn_cast_or_null" and provide an error, or is it already guaranteed by the constraints on the types in the onnx*.td file?

Reply:
Yes, it is guaranteed by the constraints in ONNXOps.td.inc.